ddsrouter_core/src/cpp/communication/Track.cpp

```diff
@@ -145,27 +145,19 @@ void Track::data_available_() noexcept
     {
         logDebug(DDSROUTER_TRACK, "Track " << *this << " has data ready to be sent.");
 
-        // Lock data_available_status_mutex_ to avoid changing the status while it is being checked
-        std::lock_guard<std::mutex> lock(data_available_status_mutex_);
+        // Get previous status and set current one to >=2 (it it was already >=2 it will keep being >2)
+        unsigned int previous_status = data_available_status_.fetch_add(DataAvailableStatus::new_data_arrived);
```

Comment:
NIT: for some humans it may be more understandable to directly use the numerical value of the enumeration.

Reply:
humans are obsolete